xray/backends/common.py

```diff
@@ -42,6 +46,28 @@ def is_trivial_index(var):
     return np.all(var.values == arange)
 
 
+def robust_getitem(array, key, catch=Exception, max_retries=6,
```

Comment:
There's also randomization in my version, which should make things work better in a multithreaded context (e.g., when using dask).